stubs/CoreGenericFunctions.phpstub

```diff
@@ -843,6 +843,17 @@ function strrchr(string $haystack, string $needle) {}
  */
 function strrev(string $string): string {}
 
+/**
+ * @psalm-pure
+ *
+ * @psalm-return (
+ *     $format is positive-int
+ *     ? array<string>
+ *     : int
+ * )
```

Comment:
Since there's a list of allowed values for `$format` we can be more precise, and flag incorrect values for the `$format` parameter as well
```suggestion
 *
 * @param 0|1|2 $format
 * @return (
 *     $format is 0 ?
 *     int : 
 *     (
 *         $format is 1 ?
 *         list<string> :
 *         array<int, string>
 *     )
 * )
```

Reply:
Thank you for the suggestion! I did not think it was necessary - but it definitely can catch some bugs.